rand_distr/src/poisson.rs

```diff
@@ -51,11 +51,7 @@ where
     Standard: Distribution<F>,
 {
     lambda: F,
```

Comment:
Is there any reason to keep the original lambda around if 1. each of the methods keeps their own lambda derivative(s) independently and 2. the original lambda is not visible to the user? If not then I would suggest removing this, and possibly even making Poisson into an enum itself (since Method will be needlessly encapsulated then)
edit: I see now that RejectionMethod also needs the original lambda. In that case I would suggest just adding `lambda` to `RejectionMethod` and removing it from here, since `KnuthMethod` doesn't need it.

Reply:
On second thought, maybe the encapsulation of `Method` would still be worthwhile to hide implementation details from the user, seeing as `Poisson` is `pub`.

Reply:
There is the issue of `Eq`. With serialization (coming from the same struct on an other machine) and some floating point nondeterminism you can get unexpected behavior. I don't think it is really something to worry about.
Also it might be nice to recover the lambda. See https://github.com/rust-random/rand/pull/1481

Reply:
Those are good points too 🤔

Reply:
Personally I think it is a mistake to implement `Eq` because it is a floating point comparison in disguise and I don't really see the use case.

Reply:
True, at most we should implement `PartialEq`.

There has been a *little* interest in adding getter methods to retrieve input parameters. I left `lambda` here to allow this, and because there doesn't appear to be any cost to doing so (enum size being determined by the largest variant).

Reply:
The lambda is in the `Poisson` struct, so it will always add to it's size

Reply:
> there doesn't appear to be any cost to doing so (enum size being determined by the largest variant)
Thanks for the reminder of that, that makes me feel better haha.
Then personally I would add lambda to each `Method` variant individually rather than keeping it in `Poisson`. To me it is preferable for `KnuthMethod` to hold more than it strictly needs (especially if variant size is not impacted) than for `RejectionMethod` to rely on `lambda` being provided from the outside when calling `sample`.